src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -50,16 +50,20 @@ public class AddPackageReferenceCommandRunner : IPackageReferenceCommandRunner
             var matchingPackageSpecs = dgSpec
                 .Projects
                 .Where(p => p.RestoreMetadata.ProjectStyle == ProjectStyle.PackageReference && 
-                PathUtility.GetStringComparerBasedOnOS().Equals(Path.GetFullPath(p.RestoreMetadata.ProjectPath), projectFullPath));
+                PathUtility.GetStringComparerBasedOnOS().Equals(Path.GetFullPath(p.RestoreMetadata.ProjectPath), projectFullPath))
+                .ToArray();
 
             // This ensures that the DG specs generated in previous steps contain exactly 1 project with the same path as the project requesting add package.
+            // Throw otherwise since we cannot proceed further.
             if (!matchingPackageSpecs.Any())
```

Comment:
nit: it is generally a good idea to use ``.Length`` instead of ``Any()`` and ``Count()`` which need to allocate enumerators to get the same thing. Not critical for this code path, but since it is an array.